### PR TITLE
ADCM-2027 Update CRUD tests for groups one more time

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

File: tests/api/__init__.py [hunk 1]
```diff
@@ -15,5 +15,4 @@
 pytestmark = [
     pytest.mark.allure_label("API Tests", label_type="layer"),
     pytest.mark.allure_label("API base", label_type="parentSuite"),
-    pytest.mark.xfail(reason="Group config will be refactored"),
 ]
```
Comment (skhomuti) [inline tests/api/__init__.py:18]: Add marks for label type `page_url`

Comment (skhomuti) [reply tests/api/__init__.py:18]: and below too

**Comment by skhomuti** [comment]
warnings `PytestCollectionWarning: cannot collect test class 'TestData' because it has a __init__ constructor (from: api/test_methods.py)` should be fixed

**Comment by vgorkavenko** [comment]
> warnings like `PytestCollectionWarning: cannot collect test class 'TestData' because it has a __init__ constructor (from: api/test_methods.py)` should be fixed

fixed

File: tests/api/__init__.py (hunk 1 above)
Comment (vgorkavenko) [reply tests/api/__init__.py:18]: done

File: tests/api/conftest.py [hunk 2]
```diff
@@ -11,6 +12,13 @@
 from tests.api.utils.api_objects import ADCMTestApiWrapper
 
 
+def pytest_configure(config: Config):
+    """
+    Exclude prepare test data classes from pytest collect
+    """
+    config.inicfg["python_classes"] = "*Test"
```
Comment (skhomuti) [inline tests/api/conftest.py:19]: how about
```
# Will not be discovered as a test
class TestClass:
    __test__ = False
```

Comment (vgorkavenko) [reply tests/api/conftest.py]: done

**Review by skhomuti — APPROVED** [review]

File: tests/api/utils/data_classes.py [hunk 3]
```diff
@@ -159,34 +163,60 @@ class ConfigLogFields(BaseClass):
...
+    description = Field(name="description", f_type=String(), default_value="auto")
+    state = Field(name="state", f_type=String(), default_value="auto")
+    url = Field(name="url", f_type=String(), default_value="auto")
+
+
+GroupConfigFields.host_candidate = Field(
+    name="host_candidate",
+    f_type=BackReferenceFK(fk_link=GroupConfigHostCandidatesFields),
+    default_value="auto",
+)
```
Comment (madlabman) [inline tests/api/utils/data_classes.py:191]: I suggest to leave a comment inside `GroupConfigFields` class declaration about this field. Or explicitly define `host_candidate` with special value as a default value (e.g. `defered` or make it obvious that field will be updated soon).

File: tests/api/testdata/bundle_community/config.yml.j2 [hunk 4]
```diff
@@ -12,34 +12,15 @@
       type: string
       default: default value
 
-  actions:
-    expand:
-      script: expand.yaml
-      script_type: ansible
```
[the remark this answers is not in the document]
Comment (vgorkavenko) [reply tests/api/testdata/bundle_community/config.yml.j2:18]: done

File: tests/api/conftest.py [hunk 5]
```diff
@@ -22,33 +23,40 @@ def pytest_addoption(parser):
...
         provider = provider_bundle.provider_prototype().provider_create(f"provider_{i}")
-        provider.host_create(fqdn=f"host_fqdn_{i}")
         cluster: Cluster = cluster_bundle.cluster_prototype().cluster_create(name=f"cluster_{i}")
-        for j in range(6)[1:]:
-            cluster.service_add(name=f"service_{j}")
+        hosts = [provider.host_create(fqdn=f"host_{provider.name}_{n}") for n in range(6)]
+        hc_list = []
+        for host in hosts:
+            cluster.host_add(host)
+            for j in range(4)[1:]:
```
[the remark this answers is not in the document]
Comment (vgorkavenko) [reply tests/api/conftest.py]: done

File: tests/api/testdata/db_filler.py [hunk 6]
```diff
@@ -98,18 +96,19 @@ def _get_or_create_data_for_endpoint(self, endpoint: Endpoints, force=False, pre
...
-        if endpoint in (Endpoints.ConfigGroup,):
-            return self._generate_data_for_special_endpoints(endpoint=endpoint, prepare_data_only=prepare_data_only)
 
         for data_class in endpoint.data_class.predefined_dependencies:
-            self._get_or_create_data_for_endpoint(endpoint=Endpoints.get_by_data_class(data_class), force=force)
+            if f"{Endpoints.get_by_data_class(data_class).path}/{{id}}" in endpoint.path:
+                endpoint_data = self._get_or_create_data_for_endpoint(endpoint=Endpoints.get_by_data_class(data_class))
+                endpoint.path = endpoint.path.format(id=str(endpoint_data[0]["id"]))
+            else:
+                self._get_or_create_data_for_endpoint(endpoint=Endpoints.get_by_data_class(data_class), force=True)
```
[the remark this answers is not in the document]
Comment (vgorkavenko) [reply tests/api/testdata/db_filler.py]: done

File: tests/api/utils/data_classes.py (hunk 3 above)
Comment (vgorkavenko) [reply tests/api/utils/data_classes.py]: done

**Review by madlabman — APPROVED** [review]

**Pull request merged**